pkg/validation/endpoints.go

```diff
@@ -0,0 +1,33 @@
+package validation
+
+import "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+// validateEndpoints checks if
+// 1. all the endpoint names are unique across components
+// 2. endpoint port are unique across component containers
+// ie; two component containers cannot have the same target port but two endpoints
+// in a single component container can have the same target port
+func validateEndpoints(endpoints []v1alpha2.Endpoint, processedEndPointPort map[int]bool, processedEndPointName map[string]bool) error {
+	currentComponentEndPointPort := make(map[int]bool)
+
+	for _, endPoint := range endpoints {
+		// Check if endpoint name is all numeric
+		if isInt(endPoint.Name) {
+			return &InvalidNameOrIdError{name: endPoint.Name, resourceType: "endpoint"}
+		}
+
+		if _, ok := processedEndPointName[endPoint.Name]; ok {
+			return &InvalidEndpointError{name: endPoint.Name}
+		}
+		processedEndPointName[endPoint.Name] = true
+		currentComponentEndPointPort[endPoint.TargetPort] = true
+	}
+
+	for targetPort := range currentComponentEndPointPort {
```

Comment:
because endpoints within a component can have the same port but across components cannot.

Valid:
component1
- endpoint1 8080
- endpoint2 8080

Invalid:
component1
-endpoint1 8080
component2
-endpoint2 8080

So by checking outside the endpoint loop, we are effectively cross checking against other components as well

Reply:
Ahh I didn't see that -- good point. Thanks!